File: c/bytecode.c
#include "bytecode.h"

int imgasm_op_length(char operation) {
    switch (operation) {
        case IMGASM_OP_NOOP:     return 0;
        case IMGASM_OP_MOVE:     return 1;
        case IMGASM_OP_RETURN:   return 0;
        case IMGASM_OP_CONSTANT: return 5;
        case IMGASM_OP_SAMPLE:   return 5;
        case IMGASM_OP_ADD:      return 1;
        case IMGASM_OP_MULTIPLY: return 1;
        default:                 return -1;
    }
}

const char* imgasm_op_name(char operation) {
    switch (operation) {
        case IMGASM_OP_NOOP:     return "NOOP";
        case IMGASM_OP_MOVE:     return "MOVE";
        case IMGASM_OP_RETURN:   return "RET ";
        case IMGASM_OP_CONSTANT: return "LOAD";
        case IMGASM_OP_SAMPLE:   return "SMPL";
        case IMGASM_OP_ADD:      return "ADD ";
        case IMGASM_OP_MULTIPLY: return "MULT";
        default:                 return "INVD";
    }
}

File: dart/lib/bytecode.dart
class OpCode {
  static const int noop = 0;
  static const int move = 2;
  static const int ret = 3;
  static const int constant = 4;
  static const int sample = 5;
  static const int add = 8;
  static const int multiply = 9;

  static int getLength(int operation) 
  {
    switch (operation) 
    {
      case noop:
      case ret:
        return 0;
      case move:
      case add:
      case multiply:
        return 1;
      case constant:
      case sample:
        return 5;
      default:
        return -1;
    }
  }

  static String getName(int operation) 
  {
    switch (operation) 
    {
      case noop: return "NOOP";
      case move: return "MOVE";
      case ret: return "RET ";
      case constant: return "LOAD";
      case sample: return "SMPL";
      case add: return "ADD ";
      case multiply: return "MULT";
      default: return "INVD";
    }
  }
}